src/Controller/NewsletterController.php

```diff
@@ -50,7 +50,7 @@ public function subscribeAction(Request $request): JsonResponse
         $email = $request->request->get('email');
         $token = $request->request->get('_token');
 
-        if(!is_string($email) || !is_string($token)){
+        if (!is_string($email) || !is_string($token)) {
```

Comment:
Why this validation isn't done as a form? It'll simplify coding and make it more modular in the future and — also — will remove token validator logic.

Reply:
Is this 'if' for $email necessary if two lines below we using ```$this->validator->validate($email)``` which check if $email is EmailType